ast/src/main/java/com/graphicsfuzz/common/tool/PrettyPrinterVisitor.java

```diff
@@ -176,6 +200,11 @@ public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration)
     visit(baseType);
     out.append(" ");
     boolean first = true;
+
+    // If the uniform value supplier provides values for this variable, emit a comment that
+    // contains the values.
+    Map<String, String> values = new HashMap<>();
+
```

Comment:
I think something like this code would be good:

```
    if (uniformValues.isPresent() && variablesDeclaration.getBaseType().hasQualifier(TypeQualifier.UNIFORM)) {
      for (VariableDeclInfo vdi : variablesDeclaration.getDeclInfos()) {
        final Optional<List<Number>> values = uniformValues.get().getValues(vdi.getName());
        values.ifPresent(item -> {
          out.append("// Contents of ")
              .append(vdi.getName())
              .append(": [")
              .append(item.stream().map(Object::toString).reduce((item1, item2) -> item1 + ", " + item2).orElse(""))
              .append("]\n");
        });
      }
    }
```

and then you can probably revert the rest of the changes to this method.

This (a) only considers uniforms, (b) considers both array and non-array uniforms.

Perhaps you could adapt it to drop the [ ] if there is only one number in the list of numbers.

Reply:
Done.